src/Word.jl

```diff
@@ -0,0 +1,68 @@
+abstract type Word{T} <: AbstractVector{T} end
+
+
+
+#interface for morphism T->Int
+function to_int{T}(t::T) end
+#interface for morphism Int->T
+function from_int{T}(t::T) end
+#interface for ordering on T
+function <=(a::T,b::T) end
+#empty string as monoid neutral element
+Base.one(::Type{Word{T}}) where {T}  = Word(Vector{T}())
+#overload for <= that implements word ordering
+function <=(a::Word{T},b::Word{T})
+#overload for * as concatenation
+function *(a::Word{T},b::Word{T}) where {T} Word(a.ptrs+b.ptrs) end
+#interface for function that returns actual vector inside word
+function data(a::Word{T})
+
+Base.length(w::Word{T}) where {T}  = length(w.ptrs)
+
+
+#Is there function composition in Julia?
+#Is it possible to assign function more or less like this?
+#In Haskell I could just make it work
+function Base.iterate(w::Word{T}) = data(w) . iterate()
```

Comment:
there is composition, but here we need to define special methods:
```julia
Base.iterate(w::Word{T}) = iterate(w.ptrs)
Base.iterate(w::Word{T}, state) = iterate(w.ptrs, state)
```
is enough

Reply:
note that it is not necessary to implement those if array interface is implemented correctly